plugin exception/FileNotFoundException on update guide

com.intellij.diagnostic.PluginException: Cannot init toolwindow org.digma.intellij.plugin.toolwindow.sidepane.DigmaSidePaneToolWindowFactory@2fe82b0f [Plugin: org.digma.intellij]
	at com.intellij.toolWindow.ToolWindowSetInitializer.createAndLayoutToolWindows$registerToolWindows(ToolWindowSetInitializer.kt:141)
	at com.intellij.toolWindow.ToolWindowSetInitializer.createAndLayoutToolWindows(ToolWindowSetInitializer.kt:160)
	at com.intellij.toolWindow.ToolWindowSetInitializer.access$createAndLayoutToolWindows(ToolWindowSetInitializer.kt:46)
	at com.intellij.toolWindow.ToolWindowSetInitializer$initUi$3.invokeSuspend(ToolWindowSetInitializer.kt:106)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:194)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:429)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:74)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:114)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:36)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:756)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:772)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.io.FileNotFoundException: /private/var/folders/xx/ylvnp1_57kj6chf_wl7nhgc00000gn/T/digma_11_guides/guides/upgrade_docker_compose.md (Permission denied)
	at java.base/java.io.FileOutputStream.open0(Native Method)
	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:184)
	at org.digma.intellij.plugin.ui.common.UpdateVersionPanel$Companion.loadFile(UpdateVersionPanel.kt:64)
	at org.digma.intellij.plugin.ui.common.UpdateVersionPanel.<init>(UpdateVersionPanel.kt:39)
	at org.digma.intellij.plugin.ui.common.MainSidePaneWindowPanelKt.createMainSidePaneWindowPanel(MainSidePaneWindowPanel.kt:15)
	at org.digma.intellij.plugin.toolwindow.sidepane.DigmaSidePaneToolWindowFactory.createToolWindowContent(DigmaSidePaneToolWindowFactory.java:59)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.createContentIfNeeded$intellij_platform_ide_impl(ToolWindowImpl.kt:543)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.scheduleContentInitializationIfNeeded$intellij_platform_ide_impl(ToolWindowImpl.kt:522)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.doShowWindow(ToolWindowManagerImpl.kt:1009)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.showToolWindowImpl(ToolWindowManagerImpl.kt:946)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.showToolWindowImpl$default(ToolWindowManagerImpl.kt:932)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.registerToolWindow$intellij_platform_ide_impl(ToolWindowManagerImpl.kt:1116)
	at com.intellij.toolWindow.ToolWindowSetInitializer.createAndLayoutToolWindows$registerToolWindows(ToolWindowSetInitializer.kt:134)
	... 34 more

--- a/src/main/java/org/digma/intellij/plugin/service/EditorService.java
+++ b/src/main/java/org/digma/intellij/plugin/service/EditorService.java
@@ -4,6 +4,7 @@ import com.intellij.icons.AllIcons;
 import com.intellij.openapi.Disposable;
 import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.ex.EditorEx;
 import com.intellij.openapi.fileEditor.FileEditorManager;
 import com.intellij.openapi.fileEditor.OpenFileDescriptor;
 import com.intellij.openapi.project.Project;
@@ -154,8 +155,14 @@ public class EditorService implements Disposable {
         return FileEditorManager.getInstance(project).openTextEditor(navigable, true);
     }
 
-    public void openVirtualFile(@NotNull VirtualFile virtualFile) {
-        openVirtualFile(virtualFile, 1);
+    public void openVirtualFile(@NotNull VirtualFile virtualFile, boolean readOnly) {
+        Editor editor = openVirtualFile(virtualFile, 1);
+        if (readOnly) {
+            if (editor instanceof EditorEx) {
+                var editorEx = (EditorEx) editor;
+                editorEx.setViewer(true);
+            }
+        }
     }
 
     private boolean showIfAlreadyOpen(String name, int lineNumber) {

--- a/src/main/kotlin/org/digma/intellij/plugin/ui/common/UpdateVersionPanel.kt
+++ b/src/main/kotlin/org/digma/intellij/plugin/ui/common/UpdateVersionPanel.kt
@@ -58,14 +58,12 @@ class UpdateVersionPanel(
             val resourceInputStream = UpdateVersionPanel.javaClass.classLoader.getResourceAsStream(resourceName)
             val ioFile = Path.of(tempDirForGuides.absolutePath, resourceName).toFile()
             FileUtil.createParentDirs(ioFile)
-            ioFile.deleteOnExit()
 
             resourceInputStream.use { input ->
                 ioFile.outputStream().use { output ->
                     input.copyTo(output)
                 }
             }
-            ioFile.setWritable(false)
 
             return LocalFileSystem.getInstance().findFileByIoFile(ioFile)!!
         }
@@ -130,11 +128,11 @@ class UpdateVersionPanel(
             if (updateState.shouldUpdateBackend) {
                 when (updateState.backendDeploymentType) {
                     BackendDeploymentType.Helm -> {
-                        EditorService.getInstance(project).openVirtualFile(updateGuideForHelm)
+                        EditorService.getInstance(project).openVirtualFile(updateGuideForHelm, true)
                     }
 
                     BackendDeploymentType.DockerCompose -> {
-                        EditorService.getInstance(project).openVirtualFile(updateGuideDockerCompose)
+                        EditorService.getInstance(project).openVirtualFile(updateGuideDockerCompose, true)
                     }
 
                     BackendDeploymentType.DockerExtension -> {

